packages/core/src/PlayerContextProvider.js

```diff
@@ -80,12 +81,23 @@ const defaultState = {
 function getGoToTrackState({
   prevState,
   index,
-  currentTime = 0,
+  track,
   shouldPlay = true,
   shouldForceLoad = false
 }) {
   const isNewTrack = prevState.activeTrackIndex !== index;
+  const currentTime = track.startingTime || 0;
+  let duration = 0;
+  if (Number.isFinite(track.duration)) {
+    duration = track.duration;
+  } else if (
+    typeof track.duration === 'string' ||
+    track.duration instanceof String
```

Comment:
Any particular reason to support the `String` constructor? I don't think we're doing that anywhere else.